### ISSUE TITLE
Fine-Tuning Bidirectional Encoder Representations From Transformers (BERT)–Based Models on Large-Scale Electronic Health Record Notes: An Empirical Study

### ISSUE THREAD

**Issue by kento1109** [kento1109/repo#32]
## タイトル
Fine-Tuning Bidirectional Encoder Representations From Transformers (BERT)–Based Models on Large-Scale Electronic Health Record Notes: An Empirical Study

## 概要
entity normalizationにBERTを適用した研究。
BERTはEHRなどのコーパスでdomain-adaptive pretrainingを行った**EhrBERT**などを用いた。

![image](https://user-images.githubusercontent.com/22322434/84554009-ee95b900-ad50-11ea-99d2-a8b40ea1faa4.png)

entity normalizationのモデル
![image](https://user-images.githubusercontent.com/22322434/84554128-7e3b6780-ad51-11ea-8c0e-f41cf23fe6f9.png)

Entityレベル（dyspnea on exertionなど）の系列を入力とし、その入力に紐づくMedDRAやSNOMED-CTなどのコードを出力する多クラス分類問題を解くことを目指す。

結果：
BERTは先行研究（MetaMapやDNorm）よりも高い精度を記録した。一方、domain-adaptive pretrainingの効果はタスクによる結果となり、必ずしもEhrBERTが優れているという結論とはならなかった。
![image](https://user-images.githubusercontent.com/22322434/84554577-77adef80-ad53-11ea-890b-af6611e958ef.png)

考察：
Entityの系列が短く、シンプルなものはnormalizationが容易だった。
また、subword分割のアドバンテージのおかげで従来予測精度が低かったOOV(Out-Of-Vocaburay)も精度が向上したと報告している。
MADEに関しては分類クラスが多いので、他のコーパスより精度が低い結果となった。

## URL
https://medinform.jmir.org/2019/3/e14830/pdf

## その他
10,000クラス以上の多クラス分類はかなり難しいのではと思ったが、F1-scoreが0.9以上という点が意外だった。だた、NCBIコーパスからMEDICへのnormalizationは、実際10,000クラスのうち、出現するのは790 classであり、実際は非常に正解ラベルが偏ったものになっていたと考えられる。（これはこの研究のlimitationという訳ではないが・・）
